test/pwave.jl

```diff
@@ -0,0 +1,58 @@
+using Test
+using Random
+using PEPSKit
+using TensorKit
+using KrylovKit
+using OptimKit
+
+"""
+    square_lattice_pwave(; t=1, μ=2, Δ=1)
+
+    Square lattice p-wave superconductor model.
+"""
+function square_lattice_pwave(; t=1, μ=2, Δ=1)
+    V = Vect[FermionParity](0 => 1, 1 => 1)
+    # on-site
+    h0 = TensorMap(zeros, ComplexF64, V ← V)
+    block(h0, FermionParity(1)) .= -μ
+    H0 = NLocalOperator{OnSite}(h0)
+    # two-site (x-direction)
+    hx = TensorMap(zeros, ComplexF64, V ⊗ V ← V ⊗ V)
+    block(hx, FermionParity(0)) .= [0 -Δ; -Δ 0]
+    block(hx, FermionParity(1)) .= [0 -t; -t 0]
+    Hx = NLocalOperator{NearestNeighbor}(hx)
+    # two-site (y-direction)
+    hy = TensorMap(zeros, ComplexF64, V ⊗ V ← V ⊗ V)
+    block(hy, FermionParity(0)) .= [0 Δ*im; -Δ*im 0]
+    block(hy, FermionParity(1)) .= [0 -t; -t 0]
+    Hy = NLocalOperator{NearestNeighbor}(hy)
+    return AnisotropicNNOperator(H0, Hx, Hy)
+end
+
+# Initialize parameters
+H = square_lattice_pwave()
+χbond = 2
+χenv = 16
+ctm_alg = CTMRG(;
+    trscheme=truncdim(χenv), tol=1e-10, miniter=4, maxiter=200, fixedspace=true, verbosity=1
+)
+opt_alg = PEPSOptimize(;
+    boundary_alg=ctm_alg,
+    optimizer=LBFGS(4; maxiter=100, gradtol=1e-3, verbosity=2),
+    gradient_alg=ManualIter(; tol),
```

Comment:
The `tol` here is not defined, causing the test to fail. Also, is there a particular reason to use `ManualIter` as the fixed-point gradient algorithm? I think `GMRES` is expected to be the 'best' approach in general, so maybe we should test using this instead?

Reply:
I think GMRES is definitely ok

Reply:
for the pwave GMRES takes significantly longer than the other options, so I choose another one

Reply:
Do we have any idea why? In principle the manual iteration should really be doing the exact same as GMRES, but GMRES has some additional steps that should make it such that it has to perform less function evaluations. Thus, I would guess that it's just some settings that have to be tweaked. Maybe the `eager=true`, or just lower the krylov subspace dimension? In any case, I would rather solve this problem than just special case this to `ManualIter` if possible :)